corehq/apps/reports/static/reports/js/tableau.js

```diff
@@ -3,12 +3,41 @@ hqDefine("reports/js/tableau", function () {
     var initialPageData = hqImport("hqwebapp/js/initial_page_data"),
         self = {};
 
-    self.initViz = function () {
+    self.requestViz = function () {
+        $.ajax({
+            method: 'post',
+            url: initialPageData.reverse('tableau_visualisation_ajax'),
+            data: {
+                validate_hostname: initialPageData.get("validate_hostname"),
+                server_name: initialPageData.get("server_address"),
+                target_site: initialPageData.get("target_site"),
+                domain_username: initialPageData.get("domain_username"),
+            },
+            dataType: 'json',
+            success: function (data) {
+                var loadingDiv = document.getElementById("loadingDiv");
+                loadingDiv.style.display = "none";
+                if (data.success) {
+                    self.initViz(data.ticket);
+                } else {
+                    document.getElementById('errorMessage').innerHTML = '<b>' + data.message + '</b>';
+                }
+            },
+        }).fail(function () {
```

Comment:
nitpick: I'd do this with the `$.ajax` `error` options instead using `fail` - it'll work the same, it just makes a nice symmetry because then the success logic and the error logic are handled with the same kind of construct.